config/default.json

```diff
@@ -130,7 +130,8 @@
         "MaxOpenConns": 300,
         "Trace": false,
         "AtRestEncryptKey": "",
-        "QueryTimeout": 30
+        "QueryTimeout": 30,
+        "ExperimentalPublicChannelsMaterialization": true
```

Comment:
Good questions! In my mind, this was less of an "experimental" feature (in that it was unfinished?), and more of a dead man's switch to give larger customers confidence that they can back out in the event their deployment behaves unexpectedly. I /think/ we want it on by default, otherwise I'm worried we'll need to recruit people to turn it on and we may languish in this state for more than a single release.

Having it on does ease the migration, but I agree we could detect it being on only from the beginning -- effectively a one-way switch. The additional complexity here is that turning it off and then on again makes the migration, itself, more complicated (since we have to deal with pre-existing data in the table). This all makes the "toggling" part of this feel less robust than the underlying changes being made.

@jasonblais & @jwilander, thoughts on whether we still want this "on by default" and whether this is truly an "experimental" feature as such?

Reply:
Maybe it can be a flag in the system table that you have to change with SQL.

Reply:
I propose we
1 - keep it in config.json, 
2 - rename it `EnablePublicChannelsMaterialization`,
3 - add it under `ServiceSettings` instead of `SqlSettings`,
4 - and set it to `true` by default.

Why?

1 - All of our other switches/flags are currently in config.json, so that is what our admins are used to. Once we move config.json configuration to the database, a database flag makes sense.

2 - As alluded to above, this is not a true "Experimental" feature. It's a feature we're confident works and improves performance for everyone. But it's a safeguard in case there's an edge case we missed and a customer needs to turn it off while we deploy a fix. Moreover, apart from performance, there are no UX changes that the setting does.

Finally, we have several other performance-related settings in config.json, which are turned on by default. They were meant to be experimental and temporary, added for Shift (given we still have them, they're not as "temporary" as we intended).

![image](https://user-images.githubusercontent.com/13119842/45244280-2b539700-b2c5-11e8-8caf-8c5df9ab4001.png)

In particular, we have at least the following:

```
        "EnablePostSearch": true,
        "EnableUserTypingMessages": true,
        "EnableChannelViewedMessages": true,
        "EnableUserStatuses": true,
```

3 - Given the other performance-related settings I alluded to above, propose we add it to that list under `ServiceSettings`.

4 - Given it's not a "true" experimental feature, propose it's enabled by default.

----

With all that said, I do realize this is a little different from the settings I listed in step 3, so feedback is welcome. But we can take care of that by clearly noting the purpose of the setting and the intention for us to remove it in our changelog and config settings doc.

Reply:
Thank you for the feedback, everyone! I'm in agreement with @jasonblais's proposal, with a minor alteration: I propose we keep it in `SqlSettings` vs. `ServiceSettings`. 

At best, this is a cosmetic change either way, but as we only pass in `SqlSettings` to the 
requisite code today, I'd have to refactor a bunch of test code to move this to another block. Since the plan is to remove it in a subsequent release, it would be ideal not to have to un-refactor this code (or leave the required modifications). 

Jason, would you be amenable to that plan?

Reply:
> I'd have to refactor a bunch of test code to move this to another block

Yeah, I think it's not worth the effort. Let's keep it in `ServiceSettings` @lieut-data